### PR TITLE
Compiling with MinGW on Windows

### PR DESCRIPTION
**Pull request by kevv87**
Here I made some hacks in order to compile libspdm as .a files on Windows.

I was able to compile the whole library with openssl, and succesfully run and passed all the library's unit tests, however the following tests were showing linking errors:
- test_size/test_size_of_spdm_responder
- test_size/test_size_of_spdm_requester

But, as said, everything else compiled fine.

Signed-off-by: Kevin Zeledon Salazar <kevin.zeledon.salazar@intel.com>

### CODE REVIEW

**Comment by jyao1** [comment]
Please your "signed-off-by" to pass DCO check in CI.

**Comment by steven-bellock** [comment]
@kevv87 You'll also want to edit https://github.com/DMTF/libspdm/blob/main/.github/workflows/build.yml to bring in MinGW and build and test libspdm using it. Probably can remove the
> - os: windows-latest
            toolchain: GCC
            
 exclusion.

File: CMakeLists.txt [hunk 1]
```diff
@@ -550,17 +593,16 @@ SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
...
                     COMMAND cp -r -f ${LIBSPDM_DIR}/unit_test/sample_key/* ${EXECUTABLE_OUTPUT_PATH})
 elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
     STRING(REPLACE "/" "\\" SRC ${LIBSPDM_DIR}/unit_test/sample_key/*)
     STRING(REPLACE "/" "\\" DEST ${EXECUTABLE_OUTPUT_PATH})
-
     ADD_CUSTOM_COMMAND(TARGET copy_sample_key
-                       PRE_BUILD
-                       COMMAND xcopy /y /s ${SRC} ${DEST})
+                    PRE_BUILD
+                    COMMAND cp -r -f ${LIBSPDM_DIR}/unit_test/sample_key/* ${EXECUTABLE_OUTPUT_PATH})
```
Comment (jyao1) [inline CMakeLists.txt:605]: you cannot use "cp" for windows.

**Comment by jyao1** [comment]
@kevv87 , thanks for the great work!
Please follow the suggestion to fix coding style CI failure.

**Comment by jyao1** [comment]
@kevv87 , test_size_of_spdm_requester and test_size_of_spdm_responder are special.

I am OK, to skip them in mingw build.

**Comment by kevv87** [comment]
Sorry for the mess, this DCO thing is new for me haha

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**